### PR TITLE
[ONNX] Merge 'initializers' into 'TorchScriptGraph'

### PR DESCRIPTION
**Pull request by BowenBao**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #95935
* #95929
* #95664
* #95651
* __->__ #95676
* #95686
* #95650

### CODE REVIEW

File: .ci/onnx/test.sh [hunk 1]
```diff
@@ -64,7 +64,7 @@ if [[ "$BUILD_ENVIRONMENT" == *onnx* ]]; then
   # TODO: change this when onnx 1.13.1 is released.
   pip install --no-use-pep517 'onnx @ git+https://github.com/onnx/onnx@e192ba01e438d22ca2dedd7956e28e3551626c91'
   # TODO: change this when onnx-script is on testPypi
-  pip install 'onnx-script @ git+https://github.com/microsoft/onnx-script@0298154caf6b46fc4e30abba034095c1290c26e3'
+  pip install 'onnx-script @ git+https://github.com/bowenbao/onnx-script@5598880b3119603dc1fcde4f164a61f6d76a1c8f'
```
Comment (BowenBao) [inline .ci/onnx/test.sh:67]: For CI validation. Change back to commit on main once merged.

**Review by titaiwangms — APPROVED** [review]
Please remember to update the onnx-script commit again when this is ready to be merged. I expect a latency.

**Comment by titaiwangms** [comment]
@BowenBao Do you think you can rearrange and we merge this one first? As it's needed to work with main branch of onnx-script.

**Comment by titaiwangms** [comment]
I think upgrade opset version to 18 would fix that error?

**Comment by BowenBao** [comment]
@titaiwangms Tried locally it is not opset 18. There is a new regression from https://github.com/microsoft/onnx-script/pull/499.

**Comment by BowenBao** [comment]
Changing the onnx-script pin to before that and trying to merge this.

**Review by wschin — APPROVED** [review]

**Pull request closed without merging**